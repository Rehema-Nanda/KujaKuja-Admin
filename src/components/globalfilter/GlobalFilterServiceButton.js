
import React from "react";

import { Button, Tooltip } from "reactstrap";
import PropTypes from "prop-types";

let buttonStyle;
let iconPath;

export default class GlobalFilterServiceButton extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            tipState: false,
            buttonState: false,
        };

        this.setTipState = this.setTipState.bind(this);
        this.setButtonState = this.setButtonState.bind(this);
    }

    componentDidMount() {
        const { selectedServiceTypes, id } = this.props;

        if (selectedServiceTypes.indexOf(id) !== -1) {
            this.setState({ buttonState: true });
        }
    }

    setTipState = () => {
        const { tipState } = this.state;

        this.setState({
            tipState: !tipState,
        });
    }

    setButtonState = (id) => {
        const { buttonState } = this.state;
        const { toggleServiceType } = this.props;

        this.setState({
            buttonState: !buttonState,
        });
        toggleServiceType(id);
    }

    render() {
        const { buttonState, tipState } = this.state;
        const {
            iconoff, iconon, tipId, id, name,
        } = this.props;

        if (!buttonState) {
            buttonStyle = "service-toggle-off";
            iconPath = iconoff;
        }
        else {
            buttonStyle = "service-toggle-on";
            iconPath = iconon;
        }

        return (

            <div>
                <Button id={tipId} data-service-id={id} className={buttonStyle} onClick={(e) => this.setButtonState(id)}><img alt="" src={iconPath} /></Button>
                <Tooltip placement="bottom" target={tipId} isOpen={tipState} toggle={this.setTipState}>{name}</Tooltip>
            </div>

        );
    }
}

GlobalFilterServiceButton.propTypes = {
    iconoff: PropTypes.string.isRequired,
    iconon: PropTypes.string.isRequired,
    tipId: PropTypes.string.isRequired,
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    toggleServiceType: PropTypes.func.isRequired,
    selectedServiceTypes: PropTypes.arrayOf(PropTypes.object).isRequired,
};
