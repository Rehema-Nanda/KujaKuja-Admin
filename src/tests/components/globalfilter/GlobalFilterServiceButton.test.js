import React from "react";
import { shallow } from "enzyme";
import GlobalFilterServiceButton from "../../../components/globalfilter/GlobalFilterServiceButton";

describe("<GlobalFilterServiceButton />", () => {
    let wrapper;
    const testProps = {
        selectedServiceTypes: ["1"],
        id: "1",
        toggleServiceType: jest.fn(),
    };
    beforeEach(() => {
        wrapper = shallow(<GlobalFilterServiceButton {...testProps} />);
    });
    it("renders correctly", () => {
        expect(wrapper.debug()).toMatchSnapshot();
    });

    it("toggles the buttton state when the button is clicked", () => {
        const button = wrapper.find("Button");
        const buttonState = wrapper.state().buttonState;
        button.simulate("click");
        expect(wrapper.state().buttonState).toEqual(!buttonState);
    });
});
