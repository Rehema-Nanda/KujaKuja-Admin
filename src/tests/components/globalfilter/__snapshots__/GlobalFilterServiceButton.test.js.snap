// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GlobalFilterServiceButton /> renders correctly 1`] = `
"<div>
  <Button id={[undefined]} data-service-id=\\"1\\" className=\\"service-toggle-on\\" onClick={[Function: onClick]} color=\\"secondary\\" tag=\\"button\\">
    <img alt=\\"\\" src={[undefined]} />
  </Button>
  <mockConstructor placement=\\"bottom\\" target={[undefined]} isOpen={false} toggle={[Function: bound ]} />
</div>"
`;
